WooCommerce/Classes/ViewRelated/Blaze/BlazeEditAd/BlazeEditAdViewModel.swift

```diff
@@ -21,33 +21,37 @@ final class BlazeEditAdViewModel: ObservableObject {
     var taglineFooterText: String {
         taglineEmptyError ?? taglineLengthLimitLabel
     }
-    private let taglineMaxLength = 32
-    @Published private var taglineRemainingLength: Int
+
+    @Published private(set) var taglineRemainingLength: Int
     private var taglineLengthLimitLabel: String {
         let lengthText = String.pluralize(taglineRemainingLength,
                                           singular: Localization.LengthLimit.singular,
                                           plural: Localization.LengthLimit.plural)
         return String(format: lengthText, taglineRemainingLength)
     }
-    private var taglineEmptyError: String?
+
+    @Published private var taglineEmptyError: String?
 
     // Description
     @Published var description: String = ""
     var descriptionFooterText: String {
         descriptionEmptyError ?? descriptionLengthLimitLabel
     }
-    private let descriptionMaxLength = 140
-    @Published private var descriptionRemainingLength: Int
+
+    @Published private(set) var descriptionRemainingLength: Int
     private var descriptionLengthLimitLabel: String {
         let lengthText = String.pluralize(descriptionRemainingLength,
                                           singular: Localization.LengthLimit.singular,
                                           plural: Localization.LengthLimit.plural)
         return String(format: lengthText, descriptionRemainingLength)
     }
-    private var descriptionEmptyError: String?
+
+    @Published private var descriptionEmptyError: String?
 
     var isSaveButtonEnabled: Bool {
-        guard let editedAdData else {
+        guard let editedAdData,
+                taglineRemainingLength >= 0,
+                descriptionRemainingLength >= 0 else {
```

Comment:
nit: Kinldly consider adding unit tests for this.

Reply:
Added tests in 444424c9cdfe5eefe3b876072ff89801007207ee.